### PR TITLE
Add plasma guns

### PR DESCRIPTION
**Pull request by r4d6**
## About The Pull Request
This add a brand new type of guns, that are high-risk, high-reward. Plasma Guns !

Directly inspired by 40k, this type of weaponry is absurdly powerful, at the cost of having to manage heat to not burn yourself, and very long reload time.

This PR do not touch the Church's plasma gun.
However it will remove Soteria's experimental plasma gun from RnD, making it admin-only.

### CORE CONCEPTS : 
--- AMMO ---
- Projectile Drop-off for each gun.
- Make the gun use non-rechargable hydrogen flasks.
- Each flask hold ~15 shots.
- Reloaded using a screwdriver to remove and secure the cell. High skill cap and take a long time.
- Failure when reloading can lead to 25 burn damage to the hand or a containment failure on the first shot.
- Backpack & Feedline can replace hydrogen flasks.

--- HEAT MECHANIC ---
- Heat the gun by firing shots.
- Lose Heat over time when not firing.
- Regularly release steam when a Heat threshold is reached, reducing Heat.
- Cannot constantly or manually vent, delay between ventings.
- Overheat by firing too fast for venting to keep up.
- When overheating, deal 25 burn damage to the active hand.
- When containment fail, give a second warning, destroy the gun and deal 50 burn damage to every bodypart.

--- PROJECTILE ---
- Deal 100 burn damage with high armor penetration.
- Does a 1-tile explosion when hitting a target.
- Maximal Mode : Use more fuel for more damage, force a delay after firing.

--- TYPES ---
- Plasma Pistol : Same Damage, less range and lower rate of fire. Can use Maximal Mode.
- Plasma Torch : Welder that use plasma flasks or backpack. Better than any other welder
- Plasma Gun : Reasonable rate of fire, can use Maximal Mode. Likely to survive through containment failure.
- Plasma Cannon : Bigger Projectile Explosion. can use Maximal Mode. 
- Plasma Incinerator : Blue Cross Weapon. ***NO RISK OF OVERHEATING OR RUNNING OUT OF AMMO***

## TODO LIST
- _**SPRITES**_.
- Testing.
- Better descriptions.

### CODE REVIEW

File: code/modules/projectiles/projectile/plasma_bullet.dm [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	impact_type = /obj/effect/projectile/plasma/impact
+	kill_count = 15
+
+/obj/item/projectile/plasma_bullet/on_impact(atom/target)
+	explosion(loc, 0, 0, 1)
+	set_light(0)
+	return TRUE
+
+/obj/item/projectile/plasma_bullet/pistol
+	kill_count = 10
```
Comment (Trilbyspaceclone) [inline code/modules/projectiles/projectile/plasma_bullet.dm:41]: 7 would be more fair

File: code/modules/projectiles/projectile/plasma_bullet.dm [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+/obj/item/projectile/plasma_bullet/pistol/welder
+	kill_count = 5
+
+/obj/item/projectile/plasma_bullet/cannon
+	kill_count = 20
```
Comment (Trilbyspaceclone) [inline code/modules/projectiles/projectile/plasma_bullet.dm:47]: 12 would be more fair

File: code/modules/projectiles/projectile/plasma_bullet.dm [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+/obj/item/projectile/plasma_bullet/cannon/on_impact(atom/target)
+	explosion(loc, 0, 1, 2, 4)
```
Comment (Trilbyspaceclone) [inline code/modules/projectiles/projectile/plasma_bullet.dm:50]: This should be lowered a small bit

File: code/modules/projectiles/projectile/plasma_bullet.dm [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+- Plasma Incinerator : Blue Cross Weapon. ***NO RISK OF OVERHEATING***, Overcharge setting is even deadlier and risk overheating. // Overcharge might not be added
+*/
+
+/obj/item/projectile/plasma_bullet
+	name = "plasma bolt"
+	icon_state = "plasma_bolt"
+	mob_hit_sound = list('sound/effects/gore/sear.ogg')
+	hitsound_wall = 'sound/weapons/guns/misc/laser_searwall.ogg'
+	damage_types = list(BURN = 100)
+	armor_penetration = 50
```
Comment (Trilbyspaceclone) [inline code/modules/projectiles/projectile/plasma_bullet.dm:29]: for sake of pvp should be made 30

File: code/modules/projectiles/projectile/plasma_bullet.dm (hunk 3 above)
Comment (r4d6) [reply code/modules/projectiles/projectile/plasma_bullet.dm:50]: It is the same value as the missile, and I have no frame of reference for explosion values.

**Pull request merged**